riffdog/resource.py

```diff
@@ -0,0 +1,77 @@
+import functools
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+
+
+class ResourceDirectory(object):
```

Comment:
Any reason why we couldn't to something similar to how Django register model admins in the admin app? https://github.com/django/django/blob/master/django/contrib/admin/sites.py#L86

I'm thinking you _might_ want to have more than one ResourceDirectory at some point (one per region perhaps?) and this would allow for that.

Reply:
I'm possibly not sure I'm following - for two reasons:

1)  the ResourceDirectory is a listing of terraform-states to process modules: whats the scenario where we would want to have different processing modules for different regions(?) (i.e. this is the block of code that can parse both a `aws_instance` and call boto to cross check - the region shouldn't change the api at boto's end?)
2) How does django register model admins? I thought it was a decorator the same as what i did, or do you mean internally to Django?


There is some debate about how 'deep' works I think, but thats possibly a futureish problem.

Reply:
Whilst writing a response to this, I realise I'm not 100% sure if it would work for this application.

For reference django admins work like this:
```python
site = AdminSite()
site.register(Model, ModelAdmin)
```
Then you can do things like:
```python
site.get_urls()
```
and it will generate urls for each ModelAdmin registered with the site.

Reply:
That kind of sounds that at that point it would bind to the `config` object, i.e. the config object gets a mapping dictionary. (where 'sites' in Django are 'configurations', i.e. this url goes to that view etc (not settings)).

Then in the state-scanner part, how does it know which config (if theres more then one) it should be looking in to do the mapping (?) - its probably a complete rewrite of that (Django I guess does url/domain/something -> site mapping... which, yes again in config we could go 'this state file (because of name, or region) -> this config', but I wonder if this is a level of complexity nobody would ever use?

One of the things that I was considering was getting rid of the config object passing and moving *that* to a singleton, which, would make the above harder.

I'm still not sure I understand the use case of multiple-different-mappings, and  thats probably my bigger problem to be able to work out if my code is 'good or bad' at this point?

Reply:
We had discussions about this and have decided to move on. @dreynolds reserves the right to claim "I told you so" if this turns out to be wrong.